pyro/contrib/forecast/forecaster.py

```diff
@@ -88,18 +88,26 @@ def predict(self, noise_dist, prediction):
             or somehow centered.
         :type noise_dist: ~pyro.distributions.Distribution
         :param prediction: A prediction for the data. This should have the same
-            shape as ``data``, but extended to full duration of the
+            shape as ``data``, but broadcastable to full duration of the
             ``covariates``.
         :type prediction: ~torch.Tensor
         """
         assert self._data is not None, ".predict() called outside .model()"
         assert self._forecast is None, ".predict() called twice"
         assert isinstance(noise_dist, dist.Distribution)
         assert isinstance(prediction, torch.Tensor)
-        assert len(noise_dist.shape()) >= 2
-        assert noise_dist.shape()[-2:] == prediction.shape[-2:]
-        assert noise_dist.event_dim in {0, 1, 2}
-        noise_dist = noise_dist.to_event(2 - noise_dist.event_dim)
+        if noise_dist.event_dim == 0:
+            if len(noise_dist.batch_shape) < 2 or noise_dist.batch_shape[-2] == 1:
```

Comment:
nit: I think we need to add `or noise_dist.batch_shape[-1] == 1` or `1 in noise_dist.batch_shape[-2:]`

Reply:
Good catch, fixed!